### ISSUE TITLE
Rate limit

### ISSUE THREAD

**Issue by baptx** [1nikolas/play-integrity-checker-app#35]
Today at around 18 CET, there was already a rate limit for the 10k tests per day and the error message said we have to wait midnight PT, which is 9 CET (so the app is unusable more than half a day).
It would be nice if users cannot easily abuse the rate limit and affect other users, by allowing around 5 requests a day per IP address.

**Comment by 1nikolas** [comment]
Yes I agree but that needs a database for the api server and I don't have the time to make that now..

**Comment by baptx** [comment]
If we just need to store IP addresses with the number of requests, I guess it would be fine to use plain text instead of a more complex database.
For example, create a text file for each IP address using the app (with the IP address as file name) and store the number of requests in the text file. Then we just need to read / write the text file and compare the number of requests stored to the one allowed per IP address. And every day, the files can be removed to reset the counter, with a cron job for example.

**Comment by 1nikolas** [comment]
but there is no server, the api is serverless hosted on vercel.com (it's free that's why i use it)

**Comment by baptx** [comment]
I saw you use a Node.js server on https://github.com/1nikolas/play-integrity-checker-server so I thought my solution would be possible.

**Comment by 1nikolas** [comment]
I use vercel for hosting which has database/kv storage but on the free plan has lots of limits so it's not an option. For the client side I don't think many people would not think to clear data and reset it. After all most people who use this app have magisk and know these basic stuff.

Maybe I can port the server into Cloudflare Workers and use [D1](https://developers.cloudflare.com/d1/) which has a generous free plan

**Comment by 1nikolas** [comment]
Fixed! Now it has 1m/day quota

**Comment by xAffan** [comment]
> Fixed! Now it has 1m/day quota

How? The pinned issue states that Google didn't approve your request? What happened?

**Comment by 1nikolas** [comment]
> > Fixed! Now it has 1m/day quota
> 
> How? The pinned issue states that Google didn't approve your request? What happened?

I asked again and they liked it this time
